sqlalchemy-stubs/sql/ddl.pyi

```diff
@@ -1,68 +1,88 @@
 from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import List
 from typing import Optional
-
-from . import roles as roles
-from .base import Executable as Executable
-from .base import SchemaVisitor as SchemaVisitor
-from .elements import ClauseElement as ClauseElement
-from .. import exc as exc
-from .. import util as util
-from ..util import topological as topological
+from typing import Sequence
+from typing import Set
+from typing import Tuple
+from typing import TypeVar
+from typing import Union
+
+from . import roles
+from .base import Executable
+from .base import SchemaVisitor
+from .elements import ClauseElement
+from .schema import ForeignKey
+from .schema import ForeignKeyConstraint
+from .schema import Index
+from .schema import SchemaItem
+from .schema import Table
+from ..engine import Connection
+from ..engine import Dialect
+from ..engine import Engine
+
+_DDLE = TypeVar("_DDLE", bound=DDLElement)
 
 class _DDLCompiles(ClauseElement): ...
 
 class DDLElement(roles.DDLRole, Executable, _DDLCompiles):
-    target: Any = ...
+    target: Optional[SchemaItem] = ...
     on: Any = ...
```

Comment:
@zzzeek is `on` ever used?

I find a reference only in `__repr__`
https://github.com/sqlalchemy/sqlalchemy/blob/0277eef8dccf34b9a715b45cc89964a8e1b46ce1/lib/sqlalchemy/sql/ddl.py#L354-L356